File: solution/210902/오픈채팅방.py
def solution(record):
    id_nickname ={}

    temp_ans = []
    answer = []

    for r in record:
        split_r = r.split(" ")

        if split_r[0] == "Enter":
            id_nickname[split_r[1]] = split_r[2]
            temp_ans.append([1, split_r[1]])
        elif split_r[0] == "Leave":
            temp_ans.append([-1, split_r[1]])
        elif split_r[0] == "Change":
            id_nickname[split_r[1]] = split_r[2]


    for a in temp_ans:
        if a[0] == 1:
            answer.append(id_nickname[a[1]]+"님이 들어왔습니다.")
        elif a[0] == -1:
            answer.append(id_nickname[a[1]]+"님이 나갔습니다.")

    return answer

File: solution/211007/오픈채팅방.java
import java.util.*;

class Solution {
    public String[] solution(String[] record) {
        HashMap<String, String> hashMap = new HashMap<>();
        List<String[]> list = new ArrayList<>();
        for (String s : record) {
            String[] split_s = s.split(" ");
            if (split_s[0].equals("Enter")){
                hashMap.put(split_s[1], split_s[2]);
                String[] ans = {"0",split_s[1]};
                list.add(ans);
            } else if(split_s[0].equals("Change")){
                hashMap.put(split_s[1], split_s[2]);
            } else if(split_s[0].equals("Leave")){
                String[] ans = {"2",split_s[1]};
                list.add(ans);
            }
        }

        List<String> answer = new ArrayList<>();

        for (String[] s : list) {
            if (s[0].equals("0")){
                answer.add(hashMap.get(s[1])+"님이 들어왔습니다.");
            } else if (s[0].equals("2")){
                answer.add(hashMap.get(s[1])+"님이 나갔습니다.");
            }
        }

        return answer.toArray(new String[0]);
    }
}
